clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -676,12 +676,12 @@ public ConsumerGroupMetadata groupMetadata() {
 
     @Override
     public void enforceRebalance() {
-        throw new KafkaException("method not implemented");
+        throw new UnsupportedOperationException("Operation not supported in new consumer group protocol");
```

Comment:
Would it be "wrong" to have the method implementation log the message instead of throwing an error?